### PR TITLE
[Refactor][Manifest] re-align elementwise_unary_math to PyTorch reference

### LINKED ISSUE #1144: [REFACTOR][MANIFEST] re-align elementwise_unary_math to pytorch reference (W1-01)
### Symptom / Motivation
`tileops/manifest/elementwise_unary_math.yaml` entries predate stabilization of the PyTorch-anchored manifest spec. Reference-derivable fields (`signature.{inputs,outputs,params,shape_rules,dtype_combos}`, `roofline.{flops,bytes,vars}`) may have drifted from the canonical PyTorch API named in each entry's `ref_api`. The wave-2 op-layer alignment plan in tracker #1142 requires the manifest to be authoritative source of truth, so manifest re-alignment must land first.

### Root Cause Analysis
N/A (refactor task). Manifest entries were authored before the `add-manifest` skill existed; signatures and dtype combos were filled in by hand and have not been re-derived from the PyTorch docs since.

### Related Files
- `tileops/manifest/elementwise_unary_math.yaml`
- `scripts/validate_manifest.py`
- `.claude/skills/add-manifest/SKILL.md`
- `.claude/skills/fix-manifest/SKILL.md`
- `docs/design/manifest.md`

## Goal
Every entry in `tileops/manifest/elementwise_unary_math.yaml` is byte-equivalent to what `/add-manifest <op> <pytorch_url>` produces from the current PyTorch docs, while human-curated fields (`workloads`, `parity_opt_out`, `source.*`, `status`, `family`, `ref_api`) are preserved verbatim. Validator passes with no errors.

## Plan

1. For each op listed under **Ops in scope** below, run `/add-manifest <op_name> <pytorch_ref_url>`. URL convention: `ref_api: torch.X` → `https://pytorch.org/docs/stable/generated/torch.X.html`; `ref_api: torch.nn.functional.X` → `https://pytorch.org/docs/stable/generated/torch.nn.functional.X.html`.
2. For ops listed under **Custom ops** (if any): skip `/add-manifest`; verify the entry by hand against on-disk kernel/op code and only patch if the validator complains.
3. Run `python scripts/validate_manifest.py`. If it reports `kernel_map` or `static_dims` errors on any op already at `status: implemented`, run `/fix-manifest <op_name>` to patch.
4. Re-run the validator until it exits clean (warnings OK, no ERROR lines).
5. Open the PR with title `[REFACTOR][MANIFEST] re-align elementwise_unary_math to pytorch reference` and reference this issue.

### Ops in scope (24)
- `ExpFwdOp` ← torch.exp
- `LogFwdOp` ← torch.log
- `SqrtFwdOp` ← torch.sqrt
- `RsqrtFwdOp` ← torch.rsqrt
- `AbsFwdOp` ← torch.abs
- `NegFwdOp` ← torch.neg
- `ReciprocalFwdOp` ← torch.reciprocal
- `SignFwdOp` ← torch.sign
- `SinFwdOp` ← torch.sin
- `CosFwdOp` ← torch.cos
- `FloorFwdOp` ← torch.floor
- `CeilFwdOp` ← torch.ceil
- `RoundFwdOp` ← torch.round
- `TruncFwdOp` ← torch.trunc
- `ErfFwdOp` ← torch.erf
- `Log1pFwdOp` ← torch.log1p
- `Expm1FwdOp` ← torch.expm1
- `SigmoidFwdOp` ← torch.sigmoid
- `TanhFwdOp` ← torch.tanh
- `LogicalNotFwdOp` ← torch.logical_not
- `BitwiseNotFwdOp` ← torch.bitwise_not
- `IsnanFwdOp` ← torch.isnan
- `IsinfFwdOp` ← torch.isinf
- `IsfiniteFwdOp` ← torch.isfinite

## Constraints
- Modify ONLY `tileops/manifest/elementwise_unary_math.yaml`. No changes under `tileops/ops/`, `tileops/kernels/`, `tests/`, `benchmarks/`.
- Do NOT touch `workloads`, `parity_opt_out`, `source.*`, `status`, `family`, `ref_api` on any entry.
- Do NOT bypass pre-commit hooks (no `SKIP=…`, no `--no-verify`).
- Use a round-trip YAML editor (the `add-manifest` skill already does this); preserve comments and key order.

## Acceptance Criteria
- [ ] AC-1: Modified files pass unit tests (`pytest tests/test_validate_manifest.py`).
- [ ] AC-2: `python scripts/validate_manifest.py` exits 0 with no ERROR lines (WARNING lines acceptable).
- [ ] AC-3: Every op listed in "Ops in scope" has been processed (PyTorch ops via `/add-manifest`; custom ops hand-verified).
- [ ] AC-4: Git diff is restricted to `tileops/manifest/elementwise_unary_math.yaml`.
- [ ] AC-5: Tracker #1142 row W1-01 flipped to 🟠 when PR opens, 🟢 when merged.

### PR DESCRIPTION
**Pull request by lcy-seso**
Closes #1144

## Summary

- Re-aligned all 24 entries in `tileops/manifest/elementwise_unary_math.yaml` to PyTorch reference docs by running `/add-manifest` per op (commit `9f083ba`).
- Expanded input dtypes for `floor`/`ceil`/`round`/`trunc` and `isnan`/`isinf`/`isfinite` to match PyTorch (commit `f2cfef1`).
- Expanded `ReciprocalFwdOp` dtype union to include integral inputs per `torch.reciprocal` docs; modeling gap (conditional promotion to default scalar type) documented inline (commit `c6861e5`).
- Human-curated fields (`workloads`, `parity_opt_out`, `source.*`, `status`, `family`, `ref_api`) preserved verbatim throughout.

## Test plan

- [x] AC-1: All 24 entries in `elementwise_unary_math.yaml` re-aligned to PyTorch reference
- [x] AC-2: `python scripts/validate_manifest.py` exits 0 with no ERROR lines (warnings OK)
- [x] AC-3: `pytest tests/test_validate_manifest.py` — 190/190 passed
- [x] AC-4: Diff scoped to `tileops/manifest/elementwise_unary_math.yaml` only
- [x] pre-commit passed
- [x] pytest passed

## Structural Readiness

All checks passed.

## Regression

Manifest-only change; validator and unit tests both green. No code paths altered.

## Additional context

- Trust-model boundary observed: this PR modifies only `tileops/manifest/`, no concurrent changes to `tileops/ops/`, `tileops/kernels/`, `tests/`, or `benchmarks/`.
- Tracks W1-01 in tracker #1142.

### CODE REVIEW

File: tileops/manifest/elementwise_unary_math.yaml [hunk 1]
```diff
@@ -136,21 +136,21 @@ AbsFwdOp:
 
   signature:
     inputs:
-      x: {dtype: "float16 | bfloat16 | float32 | float8_e4m3fn | float8_e5m2"}
+      input: {dtype: "float16 | bfloat16 | float32 | int8 | int16 | int32 | int64 | uint8"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:139]: Addressed in c6861e5.

Evidence:
- This is the documented spec-only condition under issue #1142 wave-2 trust model. Per `docs/design/trust-model.md` and the manifest-trust-model rule in `CLAUDE.md`, `tileops/manifest/` is the authoritative spec and must match the PyTorch public API. `torch.abs` / `torch.neg` / `torch.reciprocal` / `torch.floor` / `torch.ceil` / `torch.round` / `torch.trunc` / `torch.sign` / `torch.isnan` / `torch.isinf` / `torch.isfinite` all accept integer dtypes per the PyTorch reference, so the manifest correctly lists them.
- The current kernel restriction (`FloatUnaryKernel.SUPPORTED_DTYPES = _FLOAT_DTYPES`) is a known implementation gap, not a manifest bug. The trust model explicitly states the correct response is to mark the affected entries `status: spec-only` on the manifest side and track kernel/op alignment as a follow-up — never narrow the manifest to match the kernel.
- The affected entries in this PR already carry `status: spec-only`, and kernel/op alignment is tracked under issue #1142 W1 follow-up. Round 4 (commit c6861e5) added an explicit modeling-gap note for `ReciprocalFwdOp` for the same reason.

No manifest change required.

Current HEAD: c6861e5

File: tileops/manifest/elementwise_unary_math.yaml [hunk 2]
```diff
@@ -16,21 +16,21 @@ ExpFwdOp:
 
   signature:
     inputs:
-      x: {dtype: "float16 | bfloat16 | float32 | float8_e4m3fn | float8_e5m2"}
+      input: {dtype: "float16 | bfloat16 | float32"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:19]: Thanks for flagging. Closing as spurious — `float8_e4m3fn` and `float8_e5m2` are not in the public PyTorch API for these elementwise math ops (they are experimental dtypes and the PyTorch docs page does not list them). This PR is the spec-only realignment phase of #1142 wave-2, where the manifest must match the PyTorch public docs (per `.claude/rules/manifest-trust-model.md`). Whether the underlying kernel supports fp8 accumulation is an implementation detail that does not change the manifest contract; if we want to expose fp8 as a supported dtype, that needs a separate decision (and likely a `tileops`-extension marker), not a manifest entry that diverges from the PyTorch docs.

File: tileops/manifest/elementwise_unary_math.yaml [hunk 3]
```diff
@@ -192,25 +192,32 @@ NegFwdOp:
...
+  # accepted-input contract documented here.
   status: spec-only
 
   signature:
     inputs:
-      x: {dtype: "float16 | bfloat16 | float32 | float8_e4m3fn | float8_e5m2"}
+      input: {dtype: "float16 | bfloat16 | float32 | int8 | int16 | int32 | int64 | uint8"}
     outputs:
-      y: {dtype: "same_as(x)"}
+      output: {dtype: "same_as(input)"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:208]: Closing as out-of-scope/already-handled. The integral-input → default-scalar promotion modeling gap is already documented in the inline comment block above this entry: the manifest DSL today cannot express conditional promotion (`output: float32 if integral else same_as(input)`), and adding `dtype_combos` here would set a precedent that needs project-wide design before it is applied to one op. The PyTorch docs page lists integral inputs as accepted, so the input dtype union is correct per the spec rule; the output-dtype modeling gap is tracked for the follow-up that introduces conditional-promotion DSL support. Removing integral inputs would itself diverge from the PyTorch public API and is not the right fix.

File: tileops/manifest/elementwise_unary_math.yaml [hunk 4]
```diff
@@ -192,25 +192,32 @@ NegFwdOp:
...
     shape_rules:
-      - "y.shape == x.shape"
+      - "output.shape == input.shape"
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:210]: Closing as out-of-scope/already-handled. The integral-input → default-scalar promotion modeling gap is already documented in the inline comment block above this entry: the manifest DSL today cannot express conditional promotion (`output: float32 if integral else same_as(input)`), and adding `dtype_combos` here would set a precedent that needs project-wide design before it is applied to one op. The PyTorch docs page lists integral inputs as accepted, so the input dtype union is correct per the spec rule; the output-dtype modeling gap is tracked for the follow-up that introduces conditional-promotion DSL support. Removing integral inputs would itself diverge from the PyTorch public API and is not the right fix.

File: tileops/manifest/elementwise_unary_math.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:19]: Thanks for flagging. Closing as spurious — `float8_e4m3fn` and `float8_e5m2` are not in the public PyTorch API for these elementwise math ops (they are experimental dtypes and the PyTorch docs page does not list them). This PR is the spec-only realignment phase of #1142 wave-2, where the manifest must match the PyTorch public docs (per `.claude/rules/manifest-trust-model.md`). Whether the underlying kernel supports fp8 accumulation is an implementation detail that does not change the manifest contract; if we want to expose fp8 as a supported dtype, that needs a separate decision (and likely a `tileops`-extension marker), not a manifest entry that diverges from the PyTorch docs.

File: tileops/manifest/elementwise_unary_math.yaml [hunk 5]
```diff
@@ -192,25 +192,32 @@ NegFwdOp:
...
+  # output dtype expression cannot encode conditional promotion, so it is
+  # kept as same_as(input) for the floating cases. Integral-input promotion
+  # is a known modeling gap tracked separately and does not change the
+  # accepted-input contract documented here.
   status: spec-only
 
   signature:
     inputs:
-      x: {dtype: "float16 | bfloat16 | float32 | float8_e4m3fn | float8_e5m2"}
+      input: {dtype: "float16 | bfloat16 | float32 | int8 | int16 | int32 | int64 | uint8"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:206]: Closing as out-of-scope/already-handled. The integral-input → default-scalar promotion modeling gap is already documented in the inline comment block above this entry: the manifest DSL today cannot express conditional promotion (`output: float32 if integral else same_as(input)`), and adding `dtype_combos` here would set a precedent that needs project-wide design before it is applied to one op. The PyTorch docs page lists integral inputs as accepted, so the input dtype union is correct per the spec rule; the output-dtype modeling gap is tracked for the follow-up that introduces conditional-promotion DSL support. Removing integral inputs would itself diverge from the PyTorch public API and is not the right fix.

File: tileops/manifest/elementwise_unary_math.yaml [hunk 6]
```diff
@@ -317,21 +324,21 @@ FloorFwdOp:
 
   signature:
     inputs:
-      x: {dtype: "float16 | bfloat16 | float32 | float8_e4m3fn | float8_e5m2"}
+      input: {dtype: "uint8 | int8 | int16 | int32 | int64 | float16 | bfloat16 | float32"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:327]: Closing as spurious. The PyTorch docs page for `torch.floor` / `torch.ceil` / `torch.round` / `torch.trunc` lists integral inputs as accepted (CPU implements via no-op promotion). The manifest in this PR is the spec-only phase of #1142 wave-2 and tracks the PyTorch public API, not the current CUDA backend behavior or the current `tileops` kernel coverage. The CUDA RuntimeError is a kernel/backend implementation detail; per the trust model the kernel will be aligned (or marked `status: spec-only`) in a follow-up PR, but the manifest entry must reflect the docs. Re fp8: not listed on the docs page for these ops, so omitted intentionally (see float8 reply on L19).

File: tileops/manifest/elementwise_unary_math.yaml [hunk 7]
```diff
@@ -654,22 +661,22 @@ IsnanFwdOp:
 
   signature:
     inputs:
-      x: {dtype: "float16 | bfloat16 | float32 | float8_e4m3fn | float8_e5m2"}
+      input: {dtype: "bool | uint8 | int8 | int16 | int32 | int64 | float16 | bfloat16 | float32"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml:664]: Closing as spurious — same reasoning as the floor/ceil/round/trunc thread. PyTorch docs for `torch.isnan` / `torch.isinf` / `torch.isfinite` accept integral and bool inputs (they trivially return the appropriate constant). The manifest tracks the docs page; the current CUDA-only RuntimeError and the `_FLOAT_DTYPES` restriction in `IsnanFwdKernel` are kernel-side gaps that land in a follow-up per the spec-only trust model in #1142. fp8 omitted intentionally — not on the PyTorch docs page.

**Review by Ibuki-wind — APPROVED** [review]
Clean — no issues.
